src/Bicep.Core/Diagnostics/DiagnosticBuilder.cs

```diff
@@ -1232,6 +1232,86 @@ public ErrorDiagnostic UnknownModuleReferenceScheme(string badScheme, ImmutableA
                 TextSpan,
                 "BCP207",
                 $"Namespace \"{identifier}\" is imported multiple times. Remove the duplicates.");
+
+            public ErrorDiagnostic CloudProfileDoesNotExistInConfiguration(string cloudProfileName, string configurationResourceName, IEnumerable<string> availableProfileNames) => new(
+                TextSpan,
+                "BCP208",
+                $"The cloud profile \"{cloudProfileName}\" does not exist in the Bicep configuration \"{configurationResourceName}\". Available profiles include {ToQuotedString(availableProfileNames.OrderBy(x => x))}.");
+
+            public ErrorDiagnostic InvalidCloudProfileResourceManagerEndpointNullOrUndefined(string cloudProfileName, string configurationResourceName) => new(
+                TextSpan,
+                "BCP209",
+                $"The cloud profile \"{cloudProfileName}\" in the Bicep configuration \"{configurationResourceName}\" is invalid. The \"resourceManagerEndpoint\" property cannot be null or undefined.");
+
+            public ErrorDiagnostic InvalidCloudProfileInvalidResourceManagerEndpoint(string cloudProfileName, string endpoint, string configurationResourceName) => new(
+                TextSpan,
+                "BCP210",
+                $"The cloud profile \"{cloudProfileName}\" in the Bicep configuration \"{configurationResourceName}\" is invalid. The value of the \"resourceManagerEndpoint\" property is not a valid URL.");
+
+            public ErrorDiagnostic InvalidModuleAliasName(string aliasName) => new(
+                TextSpan,
+                "BCP211",
+                $"The module alias name \"{aliasName}\" is invalid. Valid characters are alphanumeric, \"_\", or \"-\".");
+
+            public ErrorDiagnostic TemplateSpecModuleAliasNameDoesNotExistInConfiguration(string aliasName, string configurationResourceName) => new(
+                TextSpan,
+                "BCP212",
+                $"The Template Spec module alias name \"{aliasName}\" does not exist in the Bicep configuration \"{configurationResourceName}\".");
+
+            public ErrorDiagnostic OciArtifactModuleAliasNameDoesNotExistInConfiguration(string aliasName, string configurationResourceName) => new(
+                TextSpan,
+                "BCP213",
+                $"The OCI artifact module alias name \"{aliasName}\" does not exist in the Bicep configuration \"{configurationResourceName}\".");
```

Comment:
>configurationResourceName [](http://example.com/codeflow?start=114&length=25)

If this is the embedded resource name in the case of default config, it will be confusing to the user.

I'm thinking we could always make it a path or a URI but make it nullable to represent built-in/default config. In these errors we could have a different text for when the path is null to make it clear to the user.

Reply:
Good suggestion. Changed `ResourceName` to a nullable `ConfigurationPath` string. For built-in config, the error message will be in the format of "...does not exist in the built-in Bicep configuration...".